functions/__tests__/util.test.ts

```diff
@@ -0,0 +1,217 @@
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bars' },
+      'docs/1'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const changeType: ChangeType = getChangeType(documentChange)
+
+    expect(changeType === ChangeType.UPDATE).toBeTruthy()
+  })
+
+  test('return a delete change type', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      {},
+      'docs/1'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const changeType: ChangeType = getChangeType(documentChange)
+
+    expect(changeType === ChangeType.DELETE).toBeTruthy()
+  })
+})
+
+describe('getDocumentId', () => {
+  const firebaseMock = firebaseFunctionsTestInit()
+
+  test('return id after create document', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      {},
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/2'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const id: string = getDocumentId(documentChange)
+
+    expect(id === '2').toBeTruthy()
+  })
+
+  test('return id after update document', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bars' },
+      'docs/2'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const id: string = getDocumentId(documentChange)
+
+    expect(id === '2').toBeTruthy()
+  })
+
+  test('return id after delete document', () => {
+    const beforeSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      { foo: 'bar' },
+      'docs/1'
+    )
+    const afterSnapshot = firebaseMock.firestore.makeDocumentSnapshot(
+      {},
+      'docs/2'
+    )
+
+    const documentChange = firebaseMock.makeChange(
+      beforeSnapshot,
+      afterSnapshot
+    )
+
+    const id: string = getDocumentId(documentChange)
+
+    expect(id === '1').toBeTruthy()
+  })
+})
+
+describe('getSearchableFields', () => {
+  let util
+  let config
+  let restoreEnv
+  let mockGetSearchableFields
+
+  beforeEach(() => {
+    jest.resetModules()
+    restoreEnv = mockedEnv(environment)
+    config = global.config
+  })
+  afterEach(() => restoreEnv())
+
+  test('configuration detected from environment variables', () => {
+    const mockConfig = config()
+    expect(mockConfig).toMatchSnapshot({})
+  })
+
+  test('return empty list', () => {
+    util = require('../src/util')
```

Comment:
does this need to be required inside the test?

Reply:
It is not elegant at all but I didn't find a way to make it work on `beforeEach()`.
It seems that it needs to load the configuration on the `beforeEach()` and after I can load the functions I test from util.